components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java

```diff
@@ -426,6 +434,104 @@ public static NativeArray jsFunction_getBusinessOwners(Context cx, Scriptable th
+            if(businessOwnerPropertiesList != null) {
+                JSONObject businessOwnerPropertiesObject = new JSONObject();
+                for (int i = 0; i < businessOwnerPropertiesList.size(); i++) {
+                    JSONObject businessOwnerPropertyObject = new JSONObject();
+                    BusinessOwnerProperty businessOwnerProperty = businessOwnerPropertiesList.get(i);
+                    businessOwnerPropertyObject.put("propertyValue", businessOwnerProperty.getPropertyValue());
+                    businessOwnerPropertyObject.put("isShowingInStore", businessOwnerProperty.isShowingInStore());
+                    businessOwnerPropertiesObject.put(businessOwnerProperty.getPropertyId(),businessOwnerPropertyObject);
+                }
+                row.put("businessOwnerProperties", row, businessOwnerPropertiesObject.toJSONString());
+            } else {
+                row.put("businessOwnerProperties", row, null);
+            }
+        } catch (NumberFormatException e) {
+            log.warn("Business owner id : " + args[0] + " is not an integer.", e);
+        }
+        return row;
+    }
+
+    /**
+     * Search business owners with pagination
+     * @param cx
+     * @param thisObj
+     * @param args
+     * @param funObj
+     * @return
+     * @throws AppManagementException
+     */
+    public static NativeObject jsFunction_searchBusinessOwners(Context cx, Scriptable thisObj, Object[] args,
+                                                           Function funObj) throws AppManagementException {
+        if (args==null||args.length != 4) {
+            handleException("Invalid number of input parameters received when searching business owners.");
+        }
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        int startIndex = Integer.parseInt(args[0].toString());
+        int pageSize = Integer.parseInt(args[1].toString());
+        int currentPage = Integer.parseInt(args[2].toString());
+        String searchValue = args[3].toString();
+        List<BusinessOwner> businessOwnerList = apiProvider.searchBusinessOwners(startIndex, pageSize, searchValue);
+        int totalBusinessOwners = apiProvider.getBusinessOwnersCount();
+        NativeObject nativeObject = new NativeObject();
+        nativeObject.put("draw", nativeObject, currentPage);
+        nativeObject.put("recordsTotal", nativeObject, totalBusinessOwners);
+        if(searchValue.trim() == "") {
+            nativeObject.put("recordsFiltered", nativeObject, totalBusinessOwners);
+        } else {
+            int filteredNoOfRecords = businessOwnerList.size();
+            nativeObject.put("recordsFiltered", nativeObject, filteredNoOfRecords);
+        }
+
+        NativeArray businessOwnersNativeArray = new NativeArray(0);
+        int businessOwnersArrayCount = 0;
+
+        for (BusinessOwner businessOwner : businessOwnerList) {
+            NativeArray businessOwnerArray = new NativeArray(0);
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/4642888b8ad6934dcda162142607b8549b0728e5